employeeApp-backend/src/main/java/io/nology/employeeAppbackend/employee/EmployeeController.java

```diff
@@ -0,0 +1,113 @@
+import org.springframework.web.server.ResponseStatusException;
+
+// Java util imports
+import java.util.List;
+import java.util.Optional;
+
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping("/employees")
+public class EmployeeController {
+    
+    Logger user = LoggerFactory.getLogger(EmployeeController.class);
+
+// Dependency injection pattern - CRUD methods
+
+    @Autowired
+    private EmployeeService service;
+
+    // Get methods
+    @CrossOrigin
+    @GetMapping
+
+    public ResponseEntity<List<Employee>> getAll() {
+    List<Employee> allEmployees = this.service.getAll();
+    user.info("Employees successfully found");
+    return new ResponseEntity<>(allEmployees, HttpStatus.OK);
+   }
+
+   @CrossOrigin
+   @GetMapping("/{id}")
+   public ResponseEntity<Employee> getById(@PathVariable Long id) {
+    Optional<Employee> employeeId = this.service.getById(id);
+        if (employeeId.isEmpty()) {
+        user.error("Employee " + id + " was not found"); 
+        return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+        }
+        user.info("Employee " + id + " found");
+        return new ResponseEntity<>(employeeId.get(), HttpStatus.OK);
+    }
+
+    // Post methods
+
+    @CrossOrigin
+    @PostMapping
+    public ResponseEntity<Employee> create(@Valid @RequestBody EmployeeDTO data) {
+        Employee createdEmployee = this.service.create(data);
+        return new ResponseEntity<Employee>(createdEmployee, HttpStatus.CREATED);
+    }
+    // Delete methods
+
+    @CrossOrigin
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Employee> delete(@PathVariable Long id) {
+        boolean isDeleted = this.service.delete(id);
+
+        if(isDeleted) {
+            return new ResponseEntity<>(null, HttpStatus.NO_CONTENT);
+        }
+
+        return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+    }
+
+     // TODO: update methods
+
+    // Update methods
+    // @PatchMapping("/{id}")
+    // public ResponseEntity<Employee> update(@RequestBody EmployeeDTO data, @PathVariable Long id) {
+    //     Optional<Employee> existingEmployee = this.service.update(data, id);
+    //     if (existingEmployee.isEmpty()) {
+    //         return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+    //     }
+    //     return this.service.update(data,id);
+
+    // }
+
+    // Update Methods
+
+    @CrossOrigin
+    @PutMapping("/{id}")
+    public ResponseEntity<Employee> update(@PathVariable Long id,       @RequestBody EmployeeUpdateDTO data) {
+        Optional<Employee> optionalEmployee = this.service.getById(id);
+        if (optionalEmployee.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "No employee found with id: " + id);
+        }
+        Employee employee = optionalEmployee.get();
+
+        Employee updatedEmployee = this.service.update(id, data, employee);
+        return new ResponseEntity<>(updatedEmployee, HttpStatus.OK);
```

Comment:
catch untracked, return 400 for startDate < endDate

Reply:
I'm getting an error when trying to implement the above - please discuss

Reply:
Updated code with if statements to catch untracked and return 400 for startDate < finishedDate

Reply:
Updated code with if statements to catch untracked and return 400 for startDate < finishedDate